### PR TITLE
Logged in user auto redirect to product

### PR DESCRIPTION
**Pull request by qiyundai**
- The logged in user will now get redirected into the product after 5s with 2s response time.
- If the user cancels the redirect sequence, the modal is suppressed for the next 3 visits
- The modal will try to get primacyCTA link from the BlockMediator, but will resort to using the querySelector to find a link that either matches the floating CTA on the page or the button with 'primacyCTA' class.
- There's some refactor done to the existing delay load product code.
- Utils proactively export helper functions and placeholders in a delayUtils package to delayed.js
- This feature will only work when all placeholders pieces are in place (cancel, bmtp-header, bmtp-cancel-text).

Resolves: [[Desktop] RTP CCX0137- Direct Path to Product](https://jira.corp.adobe.com/browse/MWPW-137828)

Test URLs:
- Before: https://stage--express--adobecom.hlx.page/express/feature/image/remove-background?martech=off
- After: https://login-auto-redirect--express--adobecom.hlx.page/express/feature/image/remove-background?martech=off

This feature requires login status to test. Might need to make its way into stage before we can do any testing. To see only the modal behavior with mocked user, please use:
https://login-auto-redirect-mock--express--adobecom.hlx.page/express/feature/image/remove-background?martech=off

### CODE REVIEW

File: express/scripts/delayed.js [hunk 1]
```diff
@@ -10,23 +10,115 @@
  * governing permissions and limitations under the License.
  */
 
-export const loadExpressProduct = async (createTag) => {
+import BlockMediator from './block-mediator.js';
```
Comment (JingleH) [inline express/scripts/delayed.js:13]: this is a bit confusing but let's use the minified BlockMediator

**Review by JingleH — CHANGES_REQUESTED** [review]

File: express/scripts/delayed.js (hunk 1 above)
Comment (qiyundai) [reply express/scripts/delayed.js:13]: Ah yes. Nice catch.

**Review by vhargrave — CHANGES_REQUESTED** [review]
Nice job! It's working pretty well.
There are a few code changes that I think need to be done to get this finished up and additionally, I'm not sure the redirect modal should stay open when I'm redirected, and navigate back in the browser: 
Screenshot 2023-10-30 at 13 56 59

File: express/scripts/delayed.js [hunk 2]
```diff
@@ -19,16 +27,99 @@ export const loadExpressProduct = async (createTag, getDevice) => {
   document.body.append(iframe);
 };
 
+async function loadLoginUserAutoRedirect() {
```
[the remark this answers is not in the document]
Comment (qiyundai) [reply express/scripts/delayed.js:30]: Sure! Gonna call it `direct-path-to-product.js` if you don't mind. I feel like `auto-redirect.js` might be a little generic for what this feature is doing.

File: express/scripts/delayed.js [hunk 3]
```diff
@@ -19,16 +27,99 @@ export const loadExpressProduct = async (createTag, getDevice) => {
...
+
+    return container;
+  };
+
+  const initRedirect = (container) => {
+    if (!followThrough) return;
+
+    container.classList.add('done');
+
+    const primaryCtaUrl = BlockMediator.get('primaryCtaUrl')
```
[the remark this answers is not in the document]
Comment (qiyundai) [reply express/scripts/delayed.js:80]: I'm not sure we need to subscribe as this is the only & last moment we'd need the `primaryCtaUrl` and we don't need to listen for any change. But please let me know if I'm missing something...

File: express/scripts/delayed.js [hunk 4]
```diff
@@ -19,16 +27,99 @@ export const loadExpressProduct = async (createTag, getDevice) => {
...
+    } else {
+      window.assign('https://new.express.adobe.com');
+    }
+  };
+
+  const profile = window.adobeProfile.getUserProfile();
+
+  const optOutCounter = localStorage.getItem('no-bmtp');
+  if (!optOutCounter || optOutCounter === '0') {
+    const container = await buildRedirectAlert(profile);
```
[the remark this answers is not in the document]
Comment (qiyundai) [reply express/scripts/delayed.js:94]: Sounds good! I'll use .then syntax!

File: express/scripts/delayed.js [hunk 5]
```diff
@@ -19,16 +27,99 @@ export const loadExpressProduct = async (createTag, getDevice) => {
...
+    setTimeout(() => {
+      if (container) initRedirect(container);
+    }, 2000);
+  }
+
+  if (optOutCounter && optOutCounter !== '0') {
+    const counterNumber = parseInt(optOutCounter, 10);
+    localStorage.setItem('no-bmtp', (counterNumber - 1).toString());
```
[the remark this answers is not in the document]
Comment (qiyundai) [reply express/scripts/delayed.js:102]: Sure!

File: express/scripts/delayed.js [hunk 6]
```diff
@@ -19,16 +27,99 @@ export const loadExpressProduct = async (createTag, getDevice) => {
...
+  }
+}
+
+export function loadDelayedLoggedIn(DELAY = 3000) {
+  return new Promise((resolve) => {
+    loadExpressProduct();
```
[the remark this answers is not in the document]
Comment (qiyundai) [reply express/scripts/delayed.js:108]: Yes. I think it is already outside the setTimeout(() => {}, DELAY) 😉

File: express/scripts/utils.js [hunk 7]
```diff
@@ -2491,8 +2491,19 @@ export async function loadArea(area = document) {
...
-  delayed([createTag, getDevice], 8000);
+
+  const { default: loadDelayed, loadDelayedLoggedIn } = await import('./delayed.js');
+  const userProfile = window.adobeProfile?.getUserProfile();
+  const placeholders = await fetchPlaceholders();
+  const autoRedirect = ['yes', 'true', 'Y', 'on'].includes(getMetadata('direct-path-to-product'));
+
+  const autoRedirectLanguageFound = placeholders.cancel || placeholders['bmtp-header'] || placeholders['bmtp-cancel-text'];
+
+  if (userProfile && autoRedirectLanguageFound && autoRedirect) {
```
[the remark this answers is not in the document]
Comment (qiyundai) [reply express/scripts/utils.js:2502]: Sure thing!

**Comment by qiyundai** [comment]
Currently waiting for Abhinav to confirm on delay second number

**Review by JingleH — APPROVED** [review]

**Review by vhargrave — APPROVED** [review]

**Pull request merged**